Code/Framework/Tests/InstanceDataHierarchy.cpp

```diff
@@ -1314,4 +1410,243 @@ namespace UnitTest
         run();
     }
 
+    TEST_F(InstanceDataHierarchyGroupTestFixture, TestNormalGroups)
+    {
+        using namespace AzToolsFramework;
+
+        // Setting up the data node hierarchy
+        AZ::SerializeContext serializeContext;
+        serializeContext.CreateEditContext();
+        Entity::Reflect(&serializeContext);
+        GroupTestComponent::Reflect(&serializeContext);
+
+        AZStd::unique_ptr<AZ::Entity> testEntity1(new AZ::Entity());
+        testEntity1->CreateComponent<GroupTestComponent>();
+
+        InstanceDataHierarchy instanceDataHierarchy;
+        instanceDataHierarchy.AddRootInstance(testEntity1.get());
+        instanceDataHierarchy.Build(&serializeContext, 0);
+
+        // Adding the nodes to a node stack
+        auto rootNode = instanceDataHierarchy.GetRootNode();
+        AZStd::stack<InstanceDataNode*> nodeStack;
+        nodeStack.push(rootNode);
+        InstanceDataNode* componentNode1 = nullptr;
+        while (!nodeStack.empty())
+        {
+            InstanceDataNode* node = nodeStack.top();
+            nodeStack.pop();
+            if (node->GetClassMetadata()->m_typeId == AZ::AzTypeInfo<GroupTestComponent>::Uuid())
+            {
+                componentNode1 = node;
+                break;
+            }
+            for (InstanceDataNode& child : node->GetChildren())
+            {
+                nodeStack.push(&child);
+            }
+        }
+        // Iterating through the children in the instance data hierarchy to verify their properties
+        ASSERT_TRUE(componentNode1 != nullptr);
```

Comment:
update: It's generally preferred to use `EXPECT_...` as opposed to ASSERT macros (so that the rest of the test can run to completion instead of failing there. There's also some great stuff you can use with Matchers to make things more succinct (I'll send you a link to some articles for reference)